egs/librispeech/ASR/conformer_ctc/asr_datamodule.py

```diff
@@ -154,70 +196,98 @@ def train_dataloaders(self) -> DataLoader:
             # to be strict (e.g. could be randomized)
             # transforms = [PerturbSpeed(factors=[0.9, 1.1], p=2/3)] + transforms   # noqa
             # Drop feats to be on the safe side.
-            cuts_train = cuts_train.drop_features()
             train = K2SpeechRecognitionDataset(
-                cuts=cuts_train,
                 cut_transforms=transforms,
                 input_strategy=OnTheFlyFeatures(
-                    Fbank(FbankConfig(num_mel_bins=80))
+                    Fbank(FbankConfig(num_mel_bins=80)),
+                    num_workers=self.args.num_workers_inner,
                 ),
                 input_transforms=input_transforms,
+                return_cuts=self.args.return_cuts,
             )
 
         if self.args.bucketing_sampler:
             logging.info("Using BucketingSampler.")
             train_sampler = BucketingSampler(
                 cuts_train,
                 max_duration=self.args.max_duration,
-                shuffle=True,
+                shuffle=self.args.shuffle,
                 num_buckets=self.args.num_buckets,
-                bucket_method='equal_duration',
+                bucket_method="equal_duration",
                 drop_last=True,
             )
         else:
             logging.info("Using SingleCutSampler.")
             train_sampler = SingleCutSampler(
                 cuts_train,
                 max_duration=self.args.max_duration,
-                shuffle=True,
+                shuffle=self.args.shuffle,
             )
         logging.info("About to create train dataloader")
-        train_dl = DataLoader(
+
+        #  train_dl = DataLoader(
+        #      train,
+        #      sampler=train_sampler,
+        #      batch_size=None,
+        #      num_workers=2,
+        #      persistent_workers=False,
+        #  )
+
+        train_dl = LhotseDataLoader(
```

Comment:
I would say be careful with `LhotseDataLoader` -- it is experimental and I'm hoping to avoid needing to use it in the future. It overcomes some I/O issues with GigaSpeech, but for LibriSpeech you shouldn't see any difference in perf with a regular DataLoader.

The downside of LhotseDataLoader is that it doesn't have the elaborate shutdown mechanisms of PyTorch DataLoader and might leave your script running after the training has finished (i.e., everything runs ok, but the script doesn't exit by itself..).